parcels/field.py

```diff
@@ -1253,47 +1225,61 @@ def __getitem__(self, key):
+    SummedField is composed of either Fields or VectorFields.
 
-    :param name: Name of the vector field
-    :param U: SummedField defining the zonal component
-    :param V: SummedField defining the meridional component
-    :param W: SummedField defining the vertical component (default: None)
+    :param name: Name of the Summed field
+    :param U: List of fields. U can be a scalar Field, a VectorField, or the zonal component of the VectorField
+    :param V: List of fields defining the meridional component (default: None)
+    :param W: List of fields defining the vertical component (default: None)
     """
 
-    def __init__(self, name, U, V, W=None):
-        self.name = name
-        self.U = U
-        self.V = V
-        self.W = W
-
-    def eval(self, time, z, y, x):
-        zonal = meridional = vertical = 0
-        if self.W is not None:
-            for (U, V, W) in zip(self.U, self.V, self.W):
-                vfld = VectorField(self.name, U, V, W)
-                vfld.fieldset = self.fieldset
-                (tmp1, tmp2, tmp3) = vfld.eval(time, z, y, x)
-                zonal += tmp1
-                meridional += tmp2
-                vertical += tmp3
-            return (zonal, meridional, vertical)
+    def __init__(self, name, U, V=None, W=None):
+        if V is None:
+            for Ui in U:
+                assert type(Ui) in [Field, VectorField], 'Components of a SummedField must be Field or VectorField'
+                self.append(Ui)
+        elif W is None:
+            for (i, Ui, Vi) in zip(range(len(U)), U, V):
+                assert isinstance(Ui, Field) and isinstance(Vi, Field), \
+                    'U, and V components of a SummedField must be Field'
+                self.append(VectorField(name+'_%d' % i, Ui, Vi))
         else:
-            for (U, V) in zip(self.U, self.V):
-                vfld = VectorField(self.name, U, V)
-                vfld.fieldset = self.fieldset
-                (tmp1, tmp2) = vfld.eval(time, z, y, x)
-                zonal += tmp1
-                meridional += tmp2
-            return (zonal, meridional)
+            for (i, Ui, Vi, Wi) in zip(range(len(U)), U, V, W):
+                assert isinstance(Ui, Field) and isinstance(Vi, Field) and isinstance(Wi, Field), \
+                    'U, V and W components of a SummedField must be Field'
+                self.append(VectorField(name+'_%d' % i, Ui, Vi, Wi))
+        self.name = name
 
     def __getitem__(self, key):
         if isinstance(key, int):
             return list.__getitem__(self, key)
         else:
-            return self.eval(*key)
+            vals = []
+            val = None
+            for iField in range(len(self)):
+                    val = list.__getitem__(self, iField).eval(*key)
+                    vals.append(val)
+            if isinstance(val, tuple):
+                return tuple(np.sum(vals, 0))
+            else:
+                return np.sum(vals)
```

Comment:
These four lines can also be written more pythonic as `return tuple(np.sum(vals, 0))  if isinstance(val, tuple) else np.sum(vals)`?